packages/aws-cdk-lib/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -87,6 +87,16 @@ export interface CommonAutoScalingGroupProps {
    */
   readonly keyName?: string;
```

Comment:
```suggestion
   * @deprecated - Use `keyPair` instead - https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_ec2-readme.html#using-an-existing-ec2-key-pair
   */
  readonly keyName?: string;
```
We should mark the property as deprecated.

Reply:
Fixed in https://github.com/aws/aws-cdk/pull/29679/commits/4940cd1a993fa6d41a7a0dde06e3feb95735a2c0